ledger/src/entry.rs

```diff
@@ -162,15 +163,73 @@ pub fn next_hash(start_hash: &Hash, num_hashes: u64, transactions: &[Transaction
     }
 }
 
+pub struct EntryVerifyState {
+    thread_h: Option<JoinHandle<()>>,
+    hashes: Option<Arc<Mutex<Vec<Hash>>>>,
+    verified: bool,
+    tx_hashes: Vec<Option<Hash>>,
+    start: Instant,
+    gpu_wait: Instant,
+}
+
+impl EntryVerifyState {
+    pub fn finish_verify(&mut self, entries: &[Entry]) -> bool {
+        if self.hashes.is_some() {
```

Comment:
this is guaranteed to be "Some" by this point, so we can just call expect() as a correctness assertion

Reply:
not for cpu